### PR TITLE
Update image dependencies

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request updates the dependencies for this image including:
- Migrate from Alpine Linux 3.18 to 3.20
- Migrate from Python 3.11.6 to 3.12.3
- Update all Python packages installed directly in the Dockerfile
- Update all Python packages installed with [pipenv]

## 💭 Motivation and context ##

Keeping dependencies up-to-date is important for project health.

## 🧪 Testing ##

Automated tests _mostly_ pass. The final image build fails because the image build exceeds the GitHub Actions runner timeout of six hours. The two most problematic platforms (`linux/ppc64le` and `linux/s390x`) will be dropped in another pull request and the build is able to complete successfully as seen in [this workflow run](https://github.com/cisagov/code-gov-update/actions/runs/10585501211).

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [ ] All new and existing tests pass.

[pipenv]: https://github.com/pypa/pipenv

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
🚀

File: src/Pipfile [hunk 1]
```diff
@@ -11,4 +11,4 @@ docopt = ">=0.6.2"
 llnl-scraper = {file = "https://github.com/cisagov/scraper/archive/v0.14.0-dev.tar.gz"}
 
 [requires]
-python_full_version = "3.10.10"
+python_full_version = "3.12.3"
```
Comment (jsf9k) [inline src/Pipfile]: Why is 3.12.3 required for this code?  This doesn't have to be in sync with the Docker container, does it?

Previously this was set to `3.10.10` while the `Dockerfile` used `3.11.6`, so I am confused as to why they have to match up now.

**Review by jsf9k — COMMENTED** [review]
I have one question.

File: src/Pipfile (hunk 1 above)
Comment (mcdonnnj) [reply src/Pipfile]: It was never updated to match the `Dockerfile` due to an oversight when merging dependabot PRs. They should always be in sync in this configuration to ensure that the Python package versions pulled by `pipenv` reflect the environment we're deploying them into.

Comment (mcdonnnj) [reply src/Pipfile]: Also this oversight is why `pipenv` installation will be done [with the following](https://github.com/cisagov/skeleton-docker/blob/ae46c280e3af456b59c271c85c95719180cb6da2/Dockerfile#L51-L52) upstream (and filter down to this project):

```dockerfile
RUN pipenv check --verbose \
    && pipenv install --clear --deploy --extra-pip-args "--no-cache-dir" --verbose
```

This _should_ fail if the Python versions do not match.

**Review by jsf9k — APPROVED** [review]

**Review by dv4harr10 — APPROVED** [review]

**Review by jsf9k — APPROVED** [review]
In the PR description, shouldn't it say "Migrate from Python 3.11.6 to 3.12._**7**_"?

**Pull request merged**